### PR TITLE
[SG-430] Modify registration component to use static ids

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
QA uses an application to run automated UI tests which require ids to locate an element. The current implementation of the registration form using the form components returns dynamic ids and this makes it difficult to run the automated tests for the UI.

## Code changes

- **apps/web/src/app/modules/register-form/register-form.component.html:** Added static ids to the form inputs.
- **libs/components/src/input/input.directive.ts:** Made the id property an input so it can be overridden when passed manually.

## Screenshots
Screenshot 2022-07-20 at 2 20 24 PM

## Before you submit

```
- [X] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Comment by vincentsalucci** [comment]
According to the @Hinton 's proposed [formatting](https://github.com/bitwarden/contributing-docs/pull/33), let's switch out all instances of camel case to use a `-` instead.

File: apps/web/src/app/modules/register-form/register-form.component.html [hunk 1]
```diff
@@ -32,12 +32,13 @@
       <bit-form-field>
         <bit-label>{{ "masterPass" | i18n }}</bit-label>
         <input
+          id="register-form_input_masterPassword"
```
Comment (Hinton) [inline apps/web/src/app/modules/register-form/register-form.component.html:35]:
```suggestion
          id="register-form_input_master-password"
```

File: apps/web/src/app/modules/register-form/register-form.component.html [hunk 2]
```diff
@@ -32,12 +32,13 @@
...
         <bit-label>{{ "masterPass" | i18n }}</bit-label>
         <input
+          id="register-form_input_master-password"
           bitInput
           (input)="updatePasswordStrength()"
           type="{{ showPassword ? 'text' : 'password' }}"
           formControlName="masterPassword"
         />
-        <button type="button" bitSuffix bitButton (click)="togglePassword()">
+        <button id="" type="button" bitSuffix bitButton (click)="togglePassword()">
```
Comment (Hinton) [inline apps/web/src/app/modules/register-form/register-form.component.html:41]: This should not have an id attribute?

Comment (gbubemismith) [reply apps/web/src/app/modules/register-form/register-form.component.html]: Thanks I must have skipped it

**Review by Hinton — APPROVED** [review]
Do you need an ID on the buttons as well? Looks good otherwise.

**Comment by gbubemismith** [comment]
> Do you need an ID on the buttons as well? Looks good otherwise.

we don't

**Pull request merged**